### PR TITLE
kanagawa-gtk-theme: init at 0-unstable-2023-07-03, kanagawa-icon-theme: init at 0-unstable-2023-07-03

### PR DESCRIPTION
**Pull request by iynaix**
## Description of changes

Adds the [Kanagwa GTK Theme](https://github.com/Fausto-Korpsvart/Kanagawa-GKT-Theme), along with its corresponding icon theme.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by donovanglover** [comment]
Should the `-gtk-theme` and `-icon-theme` be separate packages?

**Comment by iynaix** [comment]
> Should the `-gtk-theme` and `-icon-theme` be separate packages?

I was initially debating this, but after seeing someone else raise it, it seems obvious that they should be separate. I've split it into 2 packages now :smile:

File: pkgs/by-name/ka/kanagawa-gtk-theme/package.nix [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "A GTK theme with the Kanagawa colour palette";
+    homepage = "https://github.com/Fausto-Korpsvart/Kanagawa-GKT-Theme";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ iynaix ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ka/kanagawa-gtk-theme/package.nix:40]:
```suggestion
    platforms = gtk3.meta.platforms;
```

File: pkgs/by-name/ka/kanagawa-icon-theme/package.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    description = "An icon theme for the Kanagawa colour palette";
+    homepage = "https://github.com/Fausto-Korpsvart/Kanagawa-GKT-Theme";
+    license = licenses.gpl3Only;
+    maintainers = with maintainers; [ iynaix ];
+    platforms = platforms.all;
```
Comment (SuperSandro2000) [inline pkgs/by-name/ka/kanagawa-icon-theme/package.nix:45]:
```suggestion
    platforms = gtk3.meta.platforms;
```

**Review by SuperSandro2000 — APPROVED** [review]
Using gtk3.meta.platforms is probably more correct

**Comment by donovanglover** [comment]
Result of `nixpkgs-review pr 277073` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>kanagawa-gtk-theme</li>
    <li>kanagawa-icon-theme</li>
  </ul>
</details>

**Review by donovanglover — APPROVED** [review]
LGTM

**Pull request merged**